src/wp-includes/class-wp-query.php

```diff
@@ -4856,6 +4858,24 @@ protected function generate_cache_key( array $args, $sql ) {
 			$args['suppress_filters']
 		);
 
+		if ( empty( $args['post_type'] ) ) {
+			if ( $this->is_attachment ) {
+				$args['post_type'] = 'attachment';
+			} elseif ( $this->is_page ) {
+				$args['post_type'] = 'page';
+			} else {
+				$args['post_type'] = 'post';
+			}
+		}
+		$args['post_type'] = (array) $args['post_type'];
```

Comment:
I think this will cause problems with `any` and `all` post types: 

https://github.com/WordPress/wordpress-develop/blob/3b9aefb06cae1f34ed19b08caf3b3ed32df04b73/src/wp-includes/class-wp-query.php#L2536-L2544

Reply:
I agree with @peterwilsoncc. Not sure, but maybe we can do add additional checks for 'any', and use something like below:
`$args['post_type'] = get_post_types( array( 'exclude_from_search' => false ) ) ?: '';`

Reply:
Good catch @peterwilsoncc. I put in a fix here and added a unit tests. https://github.com/WordPress/wordpress-develop/pull/6393/commits/533509b7fa8465bc80eb1109144e1994c56f6a42. 

As far as I know, post type all, is not a valid argument and is not documtened anywhere, This is a case that I haven't written for. Can you give me examples in the wild where this is used and how it would break?

Reply:
I guess what is meant by all could be maybe this check if it's an array.

`! empty( $post_type ) && is_array( $post_type )`

Reply:
> I guess what is meant by all could be maybe this check if it's an array.
> 
> `! empty( $post_type ) && is_array( $post_type )`

Why is this check needed?

Reply:
> As far as I know, post type all, is not a valid argument and is not documtened anywhere, This is a case that I haven't written for. Can you give me examples in the wild where this is used and how it would break?

That's because I was thinking of `post_status` (also undocumented but used in a few examples on .org) -- sorry about that.